### PR TITLE
feat(quantity-pickers): Add error message label

### LINKED ISSUE #80: feat(quantity-picker): Add the error message param on Quantity Picker
**Is your feature request related to a problem? Please describe.**
Error message has been added to Design Specificiations.
image

**Describe the solution you'd like**
Add optional error input param to add an error message under Quantity Picker. 
Null Default value in order to have no breaking change to the component. 

**Additional context**
Closes https://github.com/Decathlon/vitamin-design/issues/30

### LINKED ISSUE #90: test(quantity-picker): Add paparazzi tests
**Describe the solution you'd like**
Write tests with paparazzi for quantity picker component for all variants & themes. 

**Additional context**
FYI Buttons already have ones.

### PR DESCRIPTION
**Pull request by corentin-stamper**
## Changes description 🧑‍💻

- Add error message
- Add paparazzi tests

## Context 🤔

Closes #80 
Closes #90 

## Checklist ✅

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch**. Please, don't request directly from your main!
- [x] Check commits & PR names matches our requested structure. It must follow the https://www.conventionalcommits.org pattern.
- [x] Check compose contract convention. It must follow conventions described [here](/CONVENTIONS.md).
- [x] Check your code additions will fail neither code linting checks.
- [x] I have reviewed the submitted code.
- [x] I have tested on a phone device/emulator.
- [x] I have tested on a tablet device/emulator.
- [x] I have tested on a large screen device/emulator.
- [x] If it includes design changes, please ask for a review `design-system-core-team-design` GitHub team.

### CODE REVIEW

File: quantity-pickers/src/main/java/com/decathlon/vitamin/compose/quantity/pickers/VitaminQuantityPickers.kt [hunk 1]
```diff
@@ -69,59 +75,98 @@ object VitaminQuantityPickers {
...
+                Icon(
+                    modifier = Modifier
+                        .size(16.dp),
+                    painter = painterResource(id = R.drawable.ic_vtmn_information_line),
+                    contentDescription = stringResource(id = R.string.vtmn_information_icon_description),
+                    tint = VitaminTheme.colors.vtmnContentNegative
                 )
-
-                MiddleTextField(
+                Box(
```
Comment (ManonPolle) [inline quantity-pickers/src/main/java/com/decathlon/vitamin/compose/quantity/pickers/VitaminQuantityPickers.kt:162]: Why this Box ?
Do you mean `Spacer` ?

File: quantity-pickers/src/main/java/com/decathlon/vitamin/compose/quantity/pickers/VitaminQuantityPickers.kt [hunk 2]
```diff
@@ -69,59 +75,98 @@ object VitaminQuantityPickers {
...
-                    shape = shape.copy(
-                        topEnd = CornerSize(0.dp),
-                        bottomEnd = CornerSize(0.dp)
-                    ),
-                    onClick = onSubtractClicked
+                Icon(
+                    modifier = Modifier
+                        .size(16.dp),
+                    painter = painterResource(id = R.drawable.ic_vtmn_information_line),
+                    contentDescription = stringResource(id = R.string.vtmn_information_icon_description),
```
Comment (ManonPolle) [inline quantity-pickers/src/main/java/com/decathlon/vitamin/compose/quantity/pickers/VitaminQuantityPickers.kt:159]: Not sure we need this content description for the icon as we already have the error text just after. Maybe we can ask @GaspardMathon or @Sabrinavigil their opinion on this.

File: quantity-pickers/src/main/java/com/decathlon/vitamin/compose/quantity/pickers/VitaminQuantityPickers.kt [hunk 3]
```diff
@@ -69,59 +75,98 @@ object VitaminQuantityPickers {
...
+            }
+        }
+    }
+
+    @Composable
+    private fun ErrorLabel(
+        errorMessage: String?
+    ) {
+        errorMessage?.let {
+            Box(
```
Comment (ManonPolle) [inline quantity-pickers/src/main/java/com/decathlon/vitamin/compose/quantity/pickers/VitaminQuantityPickers.kt:148]: Spacer() instead ?

File: quantity-pickers/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -2,4 +2,5 @@
 <resources>
     <string name="vtmn_add_button_description">Add</string>
     <string name="vtmn_subtract_button_description">Subtract</string>
+    <string name="vtmn_information_icon_description">Information</string>
```
Comment (ManonPolle) [inline quantity-pickers/src/main/res/values/strings.xml:5]: Information word to validate

**Review by ManonPolle — CHANGES_REQUESTED** [review]

**Comment by dktbitrise** [comment]
### You can download the apk here :
**sample-debug.apk :** 
https://app.bitrise.io/app/e9ae480151be4514/build/ae3d1380-7d65-44ef-af90-b3cde5772527/artifact/3e6be26b050bd4e7/p/c33dc222a16b79b7cf136f747f1b2d98

File: quantity-pickers/src/main/java/com/decathlon/vitamin/compose/quantity/pickers/VitaminQuantityPickers.kt [hunk 5]
```diff
@@ -69,59 +75,98 @@ object VitaminQuantityPickers {
...
+                ErrorLabel(errorMessage = errorMessage)
+            }
+        }
+    }
+
+    @Composable
+    private fun ErrorLabel(
+        errorMessage: String?
+    ) {
+        errorMessage?.let {
```
Comment (GerardPaligot) [inline quantity-pickers/src/main/java/com/decathlon/vitamin/compose/quantity/pickers/VitaminQuantityPickers.kt:147]: It is weird to have this check here and do thing if `errorMessage` is `null`.
Maybe you can make this check in the `Primary` component and convert `errorMessage` to no nullable parameter?

File: quantity-pickers/src/main/java/com/decathlon/vitamin/compose/quantity/pickers/VitaminQuantityPickers.kt [hunk 6]
```diff
@@ -69,59 +75,98 @@ object VitaminQuantityPickers {
...
-                    colors = colors,
-                    shape = shape.copy(
-                        topEnd = CornerSize(0.dp),
-                        bottomEnd = CornerSize(0.dp)
-                    ),
-                    onClick = onSubtractClicked
+                Icon(
+                    modifier = Modifier
+                        .size(16.dp),
+                    painter = painterResource(id = R.drawable.ic_vtmn_information_line),
```
Comment (GerardPaligot) [inline quantity-pickers/src/main/java/com/decathlon/vitamin/compose/quantity/pickers/VitaminQuantityPickers.kt:158]: /!\ If my pull request is merged before, you should migrate this to ImageVector usage.
https://github.com/Decathlon/vitamin-compose/pull/110

File: quantity-pickers/src/main/java/com/decathlon/vitamin/compose/quantity/pickers/VitaminQuantityPickers.kt [hunk 7]
```diff
@@ -69,59 +75,98 @@ object VitaminQuantityPickers {
...
-                    enabled = addEnabled,
-                    colors = colors,
-                    shape = shape.copy(
-                        topStart = CornerSize(0.dp),
-                        bottomStart = CornerSize(0.dp)
-                    ),
-                    onClick = onAddClicked
+                Text(
+                    text = it,
+                    style = VitaminTheme.typography.text3,
```
Comment (GerardPaligot) [inline quantity-pickers/src/main/java/com/decathlon/vitamin/compose/quantity/pickers/VitaminQuantityPickers.kt:168]: Please create a size class to save this typography, text field typography and icon sizes.

File: quantity-pickers/src/main/java/com/decathlon/vitamin/compose/quantity/pickers/VitaminQuantityPickers.kt [hunk 8]
```diff
@@ -69,59 +75,98 @@ object VitaminQuantityPickers {
...
+                        onClick = onSubtractClicked
                     )
+
+                    MiddleTextField(
+                        modifier = Modifier
+                            .background(
+                                color = colors.backgroundColor.copy(
+                                    alpha = if (!addEnabled && !subtractEnabled) {
+                                        VtmnStatesDisabled
+                                    } else VtmnStatesEnabled
```
Comment (GerardPaligot) [inline quantity-pickers/src/main/java/com/decathlon/vitamin/compose/quantity/pickers/VitaminQuantityPickers.kt:110]: formatting is weird here

**Review by GerardPaligot — CHANGES_REQUESTED** [review]

File: quantity-pickers/src/main/java/com/decathlon/vitamin/compose/quantity/pickers/VitaminQuantityPickers.kt (hunk 1 above)
Comment (corentin-stamper) [reply quantity-pickers/src/main/java/com/decathlon/vitamin/compose/quantity/pickers/VitaminQuantityPickers.kt:162]: Done

**Comment by dktbitrise** [comment]
### You can download the apk here :

File: quantity-pickers/src/main/java/com/decathlon/vitamin/compose/quantity/pickers/VitaminQuantityPickers.kt (hunk 2 above)
Comment (ManonPolle) [reply quantity-pickers/src/main/java/com/decathlon/vitamin/compose/quantity/pickers/VitaminQuantityPickers.kt]: Yes could be interesting to have this semantic error at QuantityPicker Level. 
It should be more useful to have an error / warning information info when you are focused on the component to update it

**Comment by dktbitrise** [comment]
### You can download the apk here :

File: quantity-pickers/README.md [hunk 9]
```diff
@@ -59,9 +59,11 @@ Parameters | Descriptions
 `subtractEnabled: (Boolean)` | Whether substract button is enable or not
 `editTextEnabled: (Boolean)` | Whether textField is enable or not
 `isExpanded: Boolean = false` | If true, component will fill max width, otherwise get default width
+`helperText: String? = null` | Helper text displayed at the QuantityPicker's bottom
 `keyboardOptions: KeyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Number)` | Software keyboard options that contains such as KeyboardType and ImeAction
 `keyboardActions: KeyboardActions = KeyboardActions.Default` | When the text input emit an IME action, the corresponding callback is called
 `colors: QuantityColors = VitaminQuantitiesColors.normal()` | The color to notify your user if they are in normal or error state
+`sizes: QuantitySizes = VitaminQuantitiesSizes.primary()` | Sizes to be applied to the QuantityPicker
 `shape: CornerBasedShape = VitaminTheme.shapes.radius100` | The shape for icon buttons
```
Comment (ManonPolle) [inline quantity-pickers/README.md:67]: remove `shape` and `textStyle`from these params as you replaced them by sizes.

File: quantity-pickers/src/main/java/com/decathlon/vitamin/compose/quantity/pickers/VitaminQuantityPickers.kt [hunk 10]
```diff
@@ -62,71 +53,53 @@ object VitaminQuantityPickers {
...
-        textStyle: TextStyle = VitaminTheme.typography.text2,
+        sizes: QuantitySizes = VitaminQuantitiesSizes.primary(),
         ripple: RippleTheme = VitaminTheme.ripples.brand,
         onAddClicked: () -> Unit,
         onSubtractClicked: () -> Unit,
         onValueChange: (String) -> Unit
     ) {
         CompositionLocalProvider(LocalRippleTheme provides ripple) {
-            Row(
+            val contentDescription = stringResource(id = R.string.vtmn_quantity_picker_accessibility_message, value)
```
Comment (ManonPolle) [inline quantity-pickers/src/main/java/com/decathlon/vitamin/compose/quantity/pickers/VitaminQuantityPickers.kt:67]: You never use your key `vtmn_quantity_picker_accessibility_message_with_helper` ?

File: quantity-pickers/src/main/res/values/strings.xml [hunk 11]
```diff
@@ -2,4 +2,7 @@
 <resources>
     <string name="vtmn_add_button_description">Add</string>
     <string name="vtmn_subtract_button_description">Subtract</string>
+
+    <string name="vtmn_quantity_picker_accessibility_message">Quantity selected: %s.</string>
+    <string name="vtmn_quantity_picker_accessibility_message_with_helper">Quantity selected: %s, %s.</string>
```
Comment (ManonPolle) [inline quantity-pickers/src/main/res/values/strings.xml:7]: never used ?

**Comment by dktbitrise** [comment]
### You can download the apk here :
**sample-debug.apk :** 
https://app.bitrise.io/app/e9ae480151be4514/build/15104298-3d6e-41c7-8d78-6a8d4e8c35b4/artifact/9ad5dcadb38501f9/p/4a160b9ab0ee4fb694422f9791d8eeb0

File: quantity-pickers/src/main/java/com/decathlon/vitamin/compose/quantity/pickers/VitaminQuantitiesColors.kt [hunk 12]
```diff
@@ -18,6 +18,8 @@ data class QuantityColors(
     val editTextBorderColor: Color,
     val focusBorderColor: Color,
     val backgroundColor: Color,
+    val helperTextColor: Color,
+    val helperIconColor: Color,
```
Comment (GerardPaligot) [inline quantity-pickers/src/main/java/com/decathlon/vitamin/compose/quantity/pickers/VitaminQuantitiesColors.kt:22]: With text input component, we have only `val helperColor: Color`. It means that the quantity picker can have an icon color different that the text? If yes, I think we need to revalidate this design specification with design team. It is weird to have a difference between text input and quantity picker for helper text.

File: quantity-pickers/src/main/java/com/decathlon/vitamin/compose/quantity/pickers/VitaminQuantityPickers.kt [hunk 13]
```diff
@@ -62,71 +53,53 @@ object VitaminQuantityPickers {
...
-                    .height(48.dp)
-                    .then(
-                        if (isExpanded) {
-                            modifier.fillMaxWidth()
-                        } else {
-                            modifier.width(156.dp)
-                        }
-                    )
+                    .semantics(
+                        mergeDescendants = true
```
Comment (GerardPaligot) [inline quantity-pickers/src/main/java/com/decathlon/vitamin/compose/quantity/pickers/VitaminQuantityPickers.kt:72]: From memory, if you configure a merge descendants, you can't access to internal components.
It means that you can't access to add and remove button with talk back?
If yes, we need to change this.

File: quantity-pickers/src/main/java/com/decathlon/vitamin/compose/quantity/pickers/VitaminQuantityPickers.kt [hunk 14]
```diff
@@ -62,71 +53,53 @@ object VitaminQuantityPickers {
...
                 modifier = modifier
-                    .height(48.dp)
-                    .then(
-                        if (isExpanded) {
-                            modifier.fillMaxWidth()
-                        } else {
-                            modifier.width(156.dp)
-                        }
-                    )
+                    .semantics(
```
Comment (GerardPaligot) [inline quantity-pickers/src/main/java/com/decathlon/vitamin/compose/quantity/pickers/VitaminQuantityPickers.kt:71]: I'm suprized you don't use `error` call inside semantic callback.
It means that you don't inform blind user when the quantity picker is in error?

File: quantity-pickers/src/main/java/com/decathlon/vitamin/compose/quantity/pickers/VitaminQuantityPickers.kt (hunk 13 above)
Comment (corentin-stamper) [reply quantity-pickers/src/main/java/com/decathlon/vitamin/compose/quantity/pickers/VitaminQuantityPickers.kt:72]: With mergeDescendants, you can access to internal components.

https://user-images.githubusercontent.com/46709142/227148835-ca227b03-96da-403b-8b6b-f35296f686b4.mp4

File: quantity-pickers/src/main/java/com/decathlon/vitamin/compose/quantity/pickers/VitaminQuantityPickers.kt (hunk 14 above)
Comment (corentin-stamper) [reply quantity-pickers/src/main/java/com/decathlon/vitamin/compose/quantity/pickers/VitaminQuantityPickers.kt:71]: Done

https://user-images.githubusercontent.com/46709142/227160080-86124c47-85a2-4c4b-bfaf-48aae25c09a8.mp4

File: quantity-pickers/src/main/java/com/decathlon/vitamin/compose/quantity/pickers/VitaminQuantitiesColors.kt (hunk 12 above)
Comment (corentin-stamper) [reply quantity-pickers/src/main/java/com/decathlon/vitamin/compose/quantity/pickers/VitaminQuantitiesColors.kt]: Validated with @MARIEDELATTRE & @Sabrinavigil 
-> Done

**Comment by dktbitrise** [comment]
### You can download the apk here :

**Comment by dktbitrise** [comment]
### You can download the apk here :
**sample-debug.apk :** 
https://app.bitrise.io/app/e9ae480151be4514/build/b4c5ca87-4e53-449f-9ad3-abe8bfd71bb5/artifact/b9196d2477026061/p/751b4e8869f2c60dd29f403d0d11be55

**Comment by dktbitrise** [comment]
### You can download the apk here :
**sample-debug.apk :** 
https://app.bitrise.io/app/e9ae480151be4514/build/f4f50772-6a14-40ee-91c0-1f89ccb2c273/artifact/857b4f3f1dfe5c71/p/30a612eabfb4dfe46cc46a8afa42bc4b

File: quantity-pickers/src/main/java/com/decathlon/vitamin/compose/quantity/pickers/internals/Picker.kt [hunk 15]
```diff
@@ -0,0 +1,105 @@
...
+    keyboardActions: KeyboardActions = KeyboardActions.Default,
+    colors: QuantityColors = VitaminQuantitiesColors.normal(),
+    sizes: QuantitySizes = VitaminQuantitiesSizes.primary(),
+    onAddClicked: () -> Unit,
+    onSubtractClicked: () -> Unit,
+    onValueChange: (String) -> Unit
+) {
+    Row(
+        modifier = Modifier
+            .height(48.dp)
```
Comment (GerardPaligot) [inline quantity-pickers/src/main/java/com/decathlon/vitamin/compose/quantity/pickers/internals/Picker.kt:48]: Move picker height to quantity picker size class

File: quantity-pickers/src/main/java/com/decathlon/vitamin/compose/quantity/pickers/internals/Picker.kt [hunk 16]
```diff
@@ -0,0 +1,105 @@
...
+            .then(
+                if (isExpanded) {
+                    modifier.fillMaxWidth()
+                } else {
+                    modifier.width(156.dp)
```
Comment (GerardPaligot) [inline quantity-pickers/src/main/java/com/decathlon/vitamin/compose/quantity/pickers/internals/Picker.kt:53]: Move picker width to quantity picker size class

File: quantity-pickers/src/main/res/values/strings.xml [hunk 17]
```diff
@@ -2,4 +2,6 @@
 <resources>
     <string name="vtmn_add_button_description">Add</string>
     <string name="vtmn_subtract_button_description">Subtract</string>
+
+    <string name="vtmn_quantity_picker_accessibility_message">Quantity selected: %s.</string>
```
Comment (GerardPaligot) [inline quantity-pickers/src/main/res/values/strings.xml:6]: Remove "." at the end

File: quantity-pickers/src/test/kotlin/com/decathlon/vitamin/compose/quantity/pickers/VitaminQuantityPickersPrimaryTest.kt [hunk 18]
```diff
@@ -0,0 +1,148 @@
...
+import com.decathlon.vitamin.compose.quantity.pickers.utils.Theme
+import com.decathlon.vitamin.compose.quantity.pickers.utils.Variant
+import com.google.testing.junit.testparameterinjector.TestParameter
+import com.google.testing.junit.testparameterinjector.TestParameterInjector
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(TestParameterInjector::class)
+internal class VitaminQuantityPickersPrimaryTest(
```
Comment (GerardPaligot) [inline quantity-pickers/src/test/kotlin/com/decathlon/vitamin/compose/quantity/pickers/VitaminQuantityPickersPrimaryTest.kt:22]: Check test results, helper text should be red

**Review by GerardPaligot — CHANGES_REQUESTED** [review]

File: quantity-pickers/src/main/java/com/decathlon/vitamin/compose/quantity/pickers/VitaminQuantityPickers.kt (hunk 6 above)
Comment (ManonPolle) [reply quantity-pickers/src/main/java/com/decathlon/vitamin/compose/quantity/pickers/VitaminQuantityPickers.kt]: It is merged, you can migrate to ImageVector

File: quantity-pickers/src/main/java/com/decathlon/vitamin/compose/quantity/pickers/internals/Picker.kt (hunk 16 above)
Comment (corentin-stamper) [reply quantity-pickers/src/main/java/com/decathlon/vitamin/compose/quantity/pickers/internals/Picker.kt]: Done

File: quantity-pickers/src/main/res/values/strings.xml (hunk 17 above)
Comment (corentin-stamper) [reply quantity-pickers/src/main/res/values/strings.xml]: Done

File: quantity-pickers/src/test/kotlin/com/decathlon/vitamin/compose/quantity/pickers/VitaminQuantityPickersPrimaryTest.kt (hunk 18 above)
Comment (corentin-stamper) [reply quantity-pickers/src/test/kotlin/com/decathlon/vitamin/compose/quantity/pickers/VitaminQuantityPickersPrimaryTest.kt:22]: Done

**Comment by dktbitrise** [comment]
### You can download the apk here :
**sample-debug.apk :** 
https://app.bitrise.io/app/e9ae480151be4514/build/bd550c48-d837-4b89-9d34-32960a5a4ab9/artifact/198aa39868375216/p/f26bb654907fd96b2b891892a38c42fc

**Review by GerardPaligot — APPROVED** [review]

**Pull request merged**